R/dittoHeatmap.R

```diff
@@ -179,7 +180,21 @@ dittoHeatmap <- function(
     all.cells <- .all_cells(object)
 
     # Make the data matrix
-    data <- as.matrix(.which_data(assay,slot,object)[genes,cells.use])
+    if (!is.null(genes)) {
+        data <- as.matrix(.which_data(assay,slot,object)[genes,cells.use])
+    } else {
+        data <- NULL
+    }
+
+    if (!is.null(metas)) {
+        met.data <- as.matrix(t(getMetas(object, names.only = FALSE)[cells.use, metas]))
+        data <- rbind(data, met.data)
+    }
+
+    if (is.null(data)) {
+        stop("No data found for given genes/metadata")
```

Comment:
The grabs will error is `genes` or `metas` are given that don't exist.  Perhaps "No 'genes' or 'metas' requested" instead